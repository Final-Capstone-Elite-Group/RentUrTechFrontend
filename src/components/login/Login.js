import React, { useState } from 'react';
import { ToastContainer } from 'react-toastify';
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { authenticateUser } from '../../redux/user/user';
import style from '../../sass/shared/form.module.scss';

const Login = () => {
  const dispatch = useDispatch();

  const [user, setUser] = useState({
    username: '',
    password: '',
  });

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(authenticateUser(user));
    setUser({
      username: '',
      password: '',
    });
  };

  return (
    <>
      <ToastContainer />
      <div className={style.wrapper}>
        <form onSubmit={handleSubmit} className={style.form}>
          <div>
            <h1>Login</h1>
            <div className={style.logo} />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="name">
              Username
            </span>
            <input type="text" id="name" name="name" value={user.username} onChange={(e) => setUser({ ...user, username: e.target.value })} required />
          </div>
          <div className={style['form-group']}>
            <span htmlFor="username">
              password
            </span>
            <input type="password" id="password" name="password" value={user.password} onChange={(e) => setUser({ ...user, password: e.target.value })} required />
          </div>
          <input type="submit" value="Log in" className={style.submit} />
          <NavLink to="/signup">signup</NavLink>
        </form>
      </div>
    </>
  );
};

export default Login;
