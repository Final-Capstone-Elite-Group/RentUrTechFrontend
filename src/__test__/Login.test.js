import { render, screen, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Login from '../components/login/Login';
import store from '../redux/configure_store';

beforeEach(() => {
  render(
    <Provider store={store}>
      <BrowserRouter>
        <Login />
      </BrowserRouter>
    </Provider>,
  );
});

test('Should match Login component snapshoot', async () => {
  const container = await waitFor(() => screen.getByText('signup'));
  expect(container).toMatchSnapshot();
});
