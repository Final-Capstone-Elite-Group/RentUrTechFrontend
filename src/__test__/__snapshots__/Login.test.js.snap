// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Login component snapshoot 1`] = `
<a
  class=""
  href="/signup"
>
  signup
</a>
`;
